src/ring_theory/subsemiring.lean

```diff
@@ -180,6 +180,10 @@ def subtype : s →+* R :=
 
 @[simp] theorem coe_subtype : ⇑s.subtype = coe := rfl
 
+/-- A subsemiring of an `ordered_semiring` is an `ordered_semiring`. -/
+instance to_ordered_semiring {R} [ordered_semiring R] (s : subsemiring R) : ordered_semiring s :=
+subtype.coe_injective.ordered_semiring coe rfl rfl (λ _ _, rfl) (λ _ _, rfl)
```

Comment:
Can you do the same for `submonoid.to_ordered_comm_monoid` and `submonoid.to_ordered_monoid`?

Reply:
> Can you do the same for `submonoid.to_ordered_comm_monoid` and `submonoid.to_ordered_monoid`? (and the to_additive versions)

Ah, this is still outstanding!  I do not have time to do it now: maybe I can try this afternoon.

Reply:
Also now missing:
* `subring.to_ordered_comm_ring`
* `subring.to_ordered_ring`

Reply:
Hopefully it should be straightforward